File: examples/ForTest.c
void n_n(int n) {
    int m = m/n;
    for(int i=1;i<m;i++)
        for(int j=1;j<=m;j++)
            k++;
}

void n_logn(int n) {
    int k = n;
    for(int i=1;i<=k+1;i++)
        for(int j=1;j<=k;j*=2)
            k++;
}

void n_2_n(int n) {
    int k = n;
    for(int i=1;i<=k;i++)
        for(int j=1;j<=1<<k;j++)
            k++;
}

void logn_n(int n) {
    int k = n;
    for(int i=1;i<=k;i*=2)
        for(int j=1;j<=k;j++)
            k++;
}

void logn_logn(int n) {
    int k = n;
    for(int i=1;i<=k;i*=2)
        for(int j=1;j<=k;j*=2)
            k++;
}
void logn_2_n(int n) {
    int k = n;
    for(int i=1;i<=k;i*=2)
        for(int j=1;j<=1<<k;j++)
            k++;
}

File: examples/ForTest.java
class Main {
	public static void n_n(int n) {
        int k = 0;
        for(int i=1;i<=n;i++)
            for(int j=1;j<=n;j++)
                k++;
    }

    public static void n_logn(int n) {
        int k = 0;
        for(int i=1;i<=n;i++)
            for(int j=1;j<=n;j*=2)
                k++;
    }

    public static void n_2n(int n) {
        int k = 0;
        for(int i=1;i<=n;i++)
            for(int j=1;j<=1<<n;j++)
                k++;
    }

    public static void logn_n(int n) {
        int k = 0;
        for(int i=1;i<=n;i*=2)
            for(int j=1;j<=n;j++)
                k++;
    }

    public static void logn_logn(int n) {
        int k = 0;
        for(int i=1;i<=n;i*=2)
            for(int j=1;j<=n;j*=2)
                k++;
    }

    public static void logn_2n(int n) {
        int k = 0;
        for(int i=1;i<=n;i*=2)
            for(int j=1;j<=1<<n;j++)
                k++;
    }

}
